### PR TITLE
[feature](array_function) add support for array_reduce

### PR DESCRIPTION
**Pull request by ZashJie**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/CMakeLists.txt [hunk 1]
```diff
@@ -87,7 +87,7 @@ message(STATUS "USE_JEMALLOC is ${USE_JEMALLOC}")
 
 # set CMAKE_BUILD_TYPE
 if (NOT CMAKE_BUILD_TYPE)
-    set(CMAKE_BUILD_TYPE RELEASE)
+    set(CMAKE_BUILD_TYPE ASAN)
```
Comment (carlvinhust2012) [inline be/CMakeLists.txt:90]: why do you change this option?

File: be/src/vec/functions/array/function_array_reduce.cpp [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+        }
+
+        std::cout << "YEAH" << std::endl;
+        
+        return arguments[1].type;
+    }
+
+    Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                        size_t result, size_t input_rows_count) override {
+        // 获取函数名
```
Comment (carlvinhust2012) [inline be/src/vec/functions/array/function_array_reduce.cpp:73]: Use English is more better.

File: be/src/vec/functions/array/function_array_reduce.cpp [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+        return arguments[1].type;
+    }
+
+    Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                        size_t result, size_t input_rows_count) override {
+        // 获取函数名                        
+        const ColumnConst& rhs_value_column =
+                static_cast<const ColumnConst&>(*block.get_by_position(arguments[0]).column.get());
+
+        // String aggregate_function_name_with_params = rhs_value_column.get_value<String>();
```
Comment (carlvinhust2012) [inline be/src/vec/functions/array/function_array_reduce.cpp:77]: why not delete the debug code?

File: be/src/vec/functions/array/function_array_reduce.cpp [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+            const auto& name = arr->get_name();
+            std::cout << &name << std::endl;
+            std::cout << "----" << std::endl;
+            // offsets_i = &arr->get_offsets();
+        }
+        else {
+            std::cout << "---------ERROR----------" << std::endl;
+        }
+
+        // if (i == 0)
```
Comment (carlvinhust2012) [inline be/src/vec/functions/array/function_array_reduce.cpp:117]: why not delete？

File: regression-test/framework/src/main/groovy/org/apache/doris/regression/suite/Suite.groovy [hunk 5]
```diff
@@ -418,6 +418,21 @@ class Suite implements GroovyInterceptable {
...
+        while (true) {
+            def stateResult = sql "show load where Label = '${label}'"
+            logger.info("load result is ${stateResult}")
+            def loadState = stateResult[stateResult.size() - 1][2].toString()
+            if ("CANCELLED".equalsIgnoreCase(loadState)) {
+                throw new IllegalStateException("load ${label} failed.")
+            } else if ("FINISHED".equalsIgnoreCase(loadState)) {
+                break
+            }
+            sleep(5000)
```
Comment (carlvinhust2012) [inline regression-test/framework/src/main/groovy/org/apache/doris/regression/suite/Suite.groovy:432]: sleep(5000) is not alway guarantee that the load is successful，may be check the return status and retry is more better.

File: be/src/vec/functions/array/function_array_reduce.cpp (hunk 4 above)
Comment (ZashJie) [reply be/src/vec/functions/array/function_array_reduce.cpp:117]: I'm sorry, I submitted a PR to have someone look at my code, so there's a lot of problematic code in there.

**Pull request closed without merging**